### DISCUSSION TITLE
Mounting static to "/"

### DISCUSSION THREAD

**Discussion by taoofshawn** [fastapi/fastapi#7461] [Questions] [answered]
### First check

* [x ] I used the GitHub search to find a similar issue and didn't find it.
* [x ] I searched the FastAPI documentation, with the integrated search.
* [x ] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

I would like the api paths to be under "/api" and the static pages loaded by the initial load at "/".  When I do this, it seems that the app.mount takes over all sub-paths and the /api paths get ignored. 

I mostly see people mounting under /static instead i.e. app.mount('/static',xxx)

How would I get the js/html front end to load and appear to be at http://server.com instead of http://server.com/static/ ?

**Comment by leonh** [comment, marked as answer]
Did you mount the api ?

Something like this works for me

```
api = FastAPI(openapi_prefix="/api")
app.mount("/api", api)
app.mount("/", StaticFiles(directory="/path/to/files/", html=True), name="static")`

```

then your api routes are decorated using api rather than app

```
@api.get("/users/me")
async def read_users_me(request: Request, user: UserDB = Depends(security)):
    print(user)
    return request.user
```

**Comment by taoofshawn** [comment]
works perfectly, thanks!

**Comment by tiangolo** [comment]
Thanks for the help here @leonh! :clap: :bow: 

Thanks for reporting back and closing the issue @taoofshawn  :+1:

**Comment by adamerose** [comment]
> Did you mount the api ?
> 
> Something like this works for me
> 
> ```
> api = FastAPI(openapi_prefix="/api")
> app.mount("/api", api)
> app.mount("/", StaticFiles(directory="/path/to/files/", html=True), name="static")`
> ```
> 
> then your api routes are decorated using api rather than app
> 
> ```
> @api.get("/users/me")
> async def read_users_me(request: Request, user: UserDB = Depends(security)):
>     print(user)
>     return request.user
> ```

For anyone else having trouble, note that the order matters here. Make sure you mount the API before the root.
